htdocs/reception/class/api_receptions.class.php

```diff
@@ -0,0 +1,721 @@
+		}
+		if(empty($proposalid)) {
+				throw new RestException(400, 'Order ID is mandatory');
+		}
+
+		$order = new Commande($this->db);
+		$result = $order->fetch($proposalid);
+		if( ! $result ) {
+				throw new RestException(404, 'Order not found');
+		}
+
+		$result = $this->reception->createFromOrder($order, DolibarrApiAccess::$user);
+		if( $result < 0) {
+				throw new RestException(405, $this->reception->error);
+		}
+		$this->reception->fetchObjectLinked();
+		return $this->_cleanObjectDatas($this->reception);
+	}
+	*/
+
+	/**
+	* Close a reception (Classify it as "Delivered")
+	*
+	* @param   int     $id             Reception ID
+	* @param   int     $notrigger      Disabled triggers
+	*
+	* @url POST    {id}/close
+	*
+	* @return  int
+	*/
+	public function close($id, $notrigger = 0)
+	{
+		if (!DolibarrApiAccess::$user->rights->reception->creer) {
+			throw new RestException(401);
+		}
+
+		$result = $this->reception->fetch($id);
+		if (!$result) {
+			throw new RestException(404, 'Reception not found');
+		}
+
+		if (!DolibarrApi::_checkAccessToResource('reception', $this->commande->id)) {
+			throw new RestException(401, 'Access not allowed for login '.DolibarrApiAccess::$user->login);
+		}
+
+		$result = $this->reception->setClosed();
+		if ($result == 0) {
+			throw new RestException(304, 'Error nothing done. May be object is already closed');
+		}
+		if ($result < 0) {
+			throw new RestException(500, 'Error when closing Order: '.$this->commande->error);
+		}
+
+		// Reload reception
+		$result = $this->reception->fetch($id);
+
+		$this->reception->fetchObjectLinked();
+
+		return $this->_cleanObjectDatas($this->reception);
+	}
+
+	// phpcs:disable PEAR.NamingConventions.ValidFunctionName.PublicUnderscore
+	/**
+	 * Clean sensible object datas
+	 *
+	 * @param   Object  $object     Object to clean
+	 * @return  Object              Object with cleaned properties
+	 */
+	protected function _cleanObjectDatas($object)
+	{
+		// phpcs:enable
+		$object = parent::_cleanObjectDatas($object);
+
+		unset($object->thirdparty); // id already returned
+
+		unset($object->note);
+		unset($object->address);
+		unset($object->barcode_type);
+		unset($object->barcode_type_code);
+		unset($object->barcode_type_label);
+		unset($object->barcode_type_coder);
+
+		if (!empty($object->lines) && is_array($object->lines)) {
+			foreach ($object->lines as $line) {
+				unset($line->tva_tx);
+				unset($line->vat_src_code);
+				unset($line->total_ht);
+				unset($line->total_ttc);
+				unset($line->total_tva);
+				unset($line->total_localtax1);
+				unset($line->total_localtax2);
+				unset($line->remise_percent);
+			}
+		}
+
+		return $object;
+	}
+
+	/**
+	 * Validate fields before create or update object
+	 *
+	 * @param   array           $data   Array with data to verify
+	 * @return  array
+	 * @throws  RestException
+	 */
+	private function _validate($data)
+	{
+		$reception = array();
+		foreach (Shipments::$FIELDS as $field) {
```

Comment:
this line is incorrect, we should replace this with Receptions::, did this and the endpoint worked